### PR TITLE
8301622: ProcessTools.java compilation gets ThreadDeath deprecation warning

### PR DESCRIPTION
**Pull request by afshin-zafari**
### Description
The use of `ThreadDeath` is replaced with checking the exception be of type `Error` and the thread is `TERMINATED`.

### Test
local and mach5 tier1

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301622](https://bugs.openjdk.org/browse/JDK-8301622): ProcessTools.java compilation gets ThreadDeath deprecation warning

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12827/head:pull/12827` \
`$ git checkout pull/12827`

Update a local copy of the PR: \
`$ git checkout pull/12827` \
`$ git pull https://git.openjdk.org/jdk pull/12827/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12827`

View PR using the GUI difftool: \
`$ git pr show -t 12827`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12827.diff">https://git.openjdk.org/jdk/pull/12827.diff</a>

</details>

### CODE REVIEW

File: test/lib/jdk/test/lib/process/ProcessTools.java [hunk 1]
```diff
@@ -823,7 +824,7 @@ static class MainThreadGroup extends ThreadGroup {
         }
 
         public void uncaughtException(Thread t, Throwable e) {
-            if (e instanceof ThreadDeath) {
+            if (e instanceof Error && t.getState() == State.TERMINATED) {
```
Comment (AlanBateman) [inline test/lib/jdk/test/lib/process/ProcessTools.java:827]: Dropping the check for ThreadDeath is fine but checking the thread state is puzzling. When a Thread completes with an uncaught exception/error then the UHE is called from the thread before it terminates so it's state won't be TERMINATED. Is there more going on here that the UHE is called with an already terminated Thread?

Comment (coleenp) [reply test/lib/jdk/test/lib/process/ProcessTools.java:827]: ok this is a strange function.  What does line 833 do?

Comment (afshin-zafari) [reply test/lib/jdk/test/lib/process/ProcessTools.java:827]: I assumed the `ThreadDeath` was there to catch `Thread.stop()` calls and just ignore them. So, I wanted to keep this functionality.
Can we remove using `ThreadDeath` relying on that `Thread.stop()` is also deprecated?

File: test/lib/jdk/test/lib/process/ProcessTools.java [hunk 2]
```diff
@@ -32,6 +32,7 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.PrintStream;
+import java.lang.Thread.State;
```
Comment (AlanBateman) [inline test/lib/jdk/test/lib/process/ProcessTools.java:35]: This is no longer needed.

**Review by dholmes-ora — APPROVED** [review]
The check for ThreadDeath was to basically ignore the uncaught exception from Thread.stop(). Such an exception is no longer possible so the check can just be deleted.

Thanks.

File: test/lib/jdk/test/lib/process/ProcessTools.java (hunk 1 above)
Comment (afshin-zafari) [reply test/lib/jdk/test/lib/process/ProcessTools.java]: Removed.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**